### ISSUE TITLE
Training failed?

### ISSUE THREAD

**Issue by alfredplpl** [PixArt-alpha/PixArt-sigma#142]
Thank you for your great code!

Currently, I'm trying to create a 0.6B Pixart-Sigma model from full scratch using 30M images. After training for several hundred thousand steps, cats still don't maintain their shape. What could be the cause of this? Possible reasons include insufficient data, insufficient parameters, or lack of text encoder capability. Which do you think it might be?
![tmp89xb0qug](https://github.com/user-attachments/assets/91d7b138-ae99-48b5-b26a-32c649af3f26)

**Comment by Feynman1999** [comment]
> Thank you for your great code!
> 
> Currently, I'm trying to create a 0.6B Pixart-Sigma model from full scratch using 30M images. After training for several hundred thousand steps, cats still don't maintain their shape. What could be the cause of this? Possible reasons include insufficient data, insufficient parameters, or lack of text encoder capability. Which do you think it might be? ![tmp89xb0qug](https://private-user-images.githubusercontent.com/3625196/354467141-91d7b138-ae99-48b5-b26a-32c649af3f26.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyNTg2NDYsIm5iZiI6MTcyMzI1ODM0NiwicGF0aCI6Ii8zNjI1MTk2LzM1NDQ2NzE0MS05MWQ3YjEzOC1hZTk5LTQ4YjUtYjI2YS0zMmM2NDlhZjNmMjYuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTBUMDI1MjI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODE5YzYwYjE5MGQ5NTI4Nzg4MzczYzg4NzkzNTZjMWM4YjZjZDg4NjJjZmM4OTM3NjIwMTEwZjdlMjNjZmRiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TvuMX9krm4HO0DkSUUcnvc5NrR0pav686rjX4567dIE)

Hello, may I ask which training script you are using?  I have tried training from scratch before, but due to limited resources, I gave up

**Comment by alfredplpl** [comment]
I fixed the training code because of the limited resources. For example, I use L4 x32 for training because I do not get A100 or H100.
The point is as follows:
- Model: bfloat16
- Optimizer: AdamW 8bit (bfloat16)
- Text Encoder: Llama-based LLM 7B 8bit (bfloat16)

We can load the model on L4 which has 24 GB VRAM.

**Comment by alfredplpl** [comment]
I continue training the model.
It seems that the cat in the inference result has a eye.

![image](https://github.com/user-attachments/assets/ba815227-eb80-48df-9880-19e0d15b2b34)

**Comment by Feynman1999** [comment]
> I continue training the model. It seems that the cat in the inference result has a eye.
> 
> ![image](https://private-user-images.githubusercontent.com/3625196/356847878-ba815227-eb80-48df-9880-19e0d15b2b34.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0NjQ1MjEsIm5iZiI6MTcyMzQ2NDIyMSwicGF0aCI6Ii8zNjI1MTk2LzM1Njg0Nzg3OC1iYTgxNTIyNy1lYjgwLTQ4ZGYtOTg4MC0xOWUwZDE1YjJiMzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTJUMTIwMzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThlN2Y3ODJlMjBmMjA4NDEzYjE4NGFjZTc0Y2MyODk5MDk4ZTAzYTE1YjRmZWExZThhYWJmOTdlYzQ1M2RhYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FmvG_B4G2HPYfbqS7i2fWJAymDFesd93Rtn1qHDz2XM)

Perhaps you can try fine-tuning the official model and training it on your dataset. I think the initial results were good, but if there were obvious cracks later on, it should be due to a bug in the training code (such as precision overflow causing gradient anomalies, data loading errors, etc.)

**Comment by alfredplpl** [comment]
I enhanced the precision. Then, I got the high quality images.

- Model: float32
- Optimizer: AdamW 8bit (bfloat16, mixed precision)
- Text Encoder: Llama-based LLM 7B 8bit (bfloat16)

![image](https://github.com/user-attachments/assets/ee28e700-73d8-4499-9b54-47daa9ac1b1a)

**Comment by alfredplpl** [comment]
![image](https://github.com/user-attachments/assets/dc0d7caf-22b6-4121-8321-10f8c7fe43dd)

I succeeded the training.
I tell you the training in detail.
Thank you.
